.github/workflows/src/context.js

```diff
@@ -246,26 +246,43 @@ export async function extractInputs(github, context, core) {
         `Could not extract build ID or project URL from check run details URL: ${checkRun.details_url}`,
       );
     }
-    if (
-      !context.payload.repository ||
-      !context.payload.repository.owner ||
-      !context.payload.repository.owner.login ||
-      !context.payload.repository.name
-    ) {
-      throw new Error(
-        `Could not extract repository owner or name from context payload: ${JSON.stringify(context.payload.repository)}`,
+
+    const payload =
+      /** @type {import("@octokit/webhooks-types").CheckRunEvent} */ (
+        context.payload
       );
-    }
+const repositoryInfo = getRepositoryInfo(payload.repository);
     inputs = {
-      owner: context.payload.repository.owner.login,
-      repo: context.payload.repository.name,
+      owner: repositoryInfo.owner,
+      repo: repositoryInfo.repo,
       head_sha: checkRun.head_sha,
       ado_build_id: match[2],
       ado_project_url: match[1],
       issue_number: NaN,
       run_id: NaN,
     };
+  } else if (
+    context.eventName === "check_suite" &&
+    context.payload.action === "completed"
+  ) {
+    const payload =
+      /** @type {import("@octokit/webhooks-types").CheckSuiteCompletedEvent} */ (
+        context.payload
+      );
+
+    inputs = {
+      ...getRepositoryInfo(payload.repository),
+      head_sha: payload.check_suite.head_sha,
+
+      // These are NaN today because the only consumer of this event needs only
+      // the head_sha
+      issue_number: NaN,
+      run_id: NaN,
+    };
   } else {
+    core.debug("context.payload");
+    core.debug(JSON.stringify(context.payload));
```

Comment:
We already log this above:

https://github.com/Azure/azure-rest-api-specs/pull/33806/files#diff-c5443c116935f574137c66a964701bbaacda79bb952dac85bff4947b866c540dR25